test/integration/react-18/prerelease/components/bar.js

```diff
@@ -0,0 +1,24 @@
+import { Suspense } from 'react'
+import dynamic from 'next/dynamic'
+
+const Foo = dynamic(() => import('./foo'), {
+  suspense: true,
+  loadableGenerated: {
+    modules: ['./foo'],
+    webpack: [require.resolveWeak('./foo')],
+  },
+})
+
+export default function Bar() {
+  if (typeof window === 'undefined') {
+    throw new Promise((resolve) => setTimeout(resolve, 300))
+  }
```

Comment:
I'm not sure how much it matters at the moment, but this isn't technically valid -- you should throw the same promise.